### PR TITLE
Update README.md and Requirements.txt

### PR DESCRIPTION
**Pull request by mofosyne**
Add to readme information about mathplotlib

### CODE REVIEW

**Comment by mofosyne** [comment]
This change was based on my experience in this issue ticket https://github.com/fablabnbg/inkscape-silhouette/issues/245

This readme update doesn't address the udev rule issue encountered in it as I'm not sure what the best way to explain why it doesn't work on a fresh install and that you have to add extra udev rules to make this work.

**Comment by mofosyne** [comment]
Btw... I've been wondering how to get a console log of this extension so I can better understand what it's thinking.

Also bit of a python noob so would be good to hear recommendation how to hook up a debugger to this.

**Comment by t0b3** [comment]
> Btw... I've been wondering how to get a console log of this extension so I can better understand what it's thinking.
> 
> Also bit of a python noob so would be good to hear recommendation how to hook up a debugger to this.

Simply run `python sendto_silhouette.py --help` or copy the specific command from any of your log files.
You will get some output on cmdline and even more in the log file. Also consider `--logfile /tmp/PATH_OF_YOUR_CHOICE_TO_LOGFILE`

**Comment by t0b3** [comment]
> Add to readme information about matplotlib

Thank you for your valuable contribution. Dependencies may change every now and then. (i.e. Inscape 1.4 will require `tinycss2`) To avoid duplication and sync' efforts I'd rather prefer `pip install -r requirements.txt`. This also helps keeping instructions more platform agnostic (except for GTK and matplotlib which are better managed by the distro). Can you update this PR accordingly?

BTW: And at the same time I'd like to merge the two Ubuntu parts keeping the inkscape:stable repo instruction with the advantage of using the latest version. (currently 1.3)

**Comment by mofosyne** [comment]
> Simply run python sendto_silhouette.py --help or copy the specific command from any of your log files.
You will get some output on cmdline and even more in the log file. Also consider --logfile /tmp/PATH_OF_YOUR_CHOICE_TO_LOGFILE

That would require me to run that script outside of inkscape? Might be a bit clunky.

I think what would be ideal is to set `/tmp/inkscape-silhouette.log` as the logfile by default. Not sure where to add that in however.

**Comment by t0b3** [comment]
> > Simply run python sendto_silhouette.py --help or copy the specific command from any of your log files.
> > You will get some output on cmdline and even more in the log file. Also consider --logfile /tmp/PATH_OF_YOUR_CHOICE_TO_LOGFILE
> 
> That would require me to run that script outside of inkscape? Might be a bit clunky.
> 
> I think what would be ideal is to set `/tmp/inkscape-silhouette.log` as the logfile by default. Not sure where to add that in however.

While there are use cases that invoke `inkscape-silhouette.py` directly, yet it's obvious this won't fit for all.

**Comment by mofosyne** [comment]
This readme change is ready to merge now

Regarding logging. Created a separate PR for it https://github.com/fablabnbg/inkscape-silhouette/pull/249 "sento_silhouette: logfile param defaults to /tmp/inkscape-silhouette.log #249" so that it will pick a system default location for the logfile if empty and enabled.

File: README.md [hunk 1]
```diff
@@ -28,37 +28,50 @@ This extension should work with the following devices:
 
 ## Installation
 
-### Ubuntu 20.10 or newer (features Inkscape 1.0+ natively)
+### Ubuntu
 
-WARNING: SNAP packages may cause issues.
+WARNING: SNAP packages may cause issues. We use deb file shown later in this section.
 
-Install Inkscape
```
Comment (t0b3) [inline README.md:35]: `#### Install Inkscape and other requirements`

File: README.md [hunk 2]
```diff
@@ -28,37 +28,50 @@ This extension should work with the following devices:
...
-* `sudo apt install inkscape`
+# Install all requirements from python package manager
+sudo apt-get install python3-pip
+python3 -m pip install -U pip
+python3 -m pip install -r requirements.txt
+```
+</details>
 
-Install requirements
```
Comment (t0b3) [inline README.md:56]: `#### Install inkscape-silhouette`

File: README.md [hunk 3]
```diff
@@ -28,37 +28,50 @@ This extension should work with the following devices:
...
+So now we shall install inkscape-silhouette, so scroll down the latest releases and head to the Assets section of releases and click on the *.deb file. You can then use `sudo apt-get install ./*.deb` where `*.deb` is the name of your newly downloaded file.
 
-Install inkscape-silhouette
 * https://github.com/fablabnbg/inkscape-silhouette/releases
-  Scroll down to Downloads and click on the *.deb file.
+
+<details>
+<summary>Maintainers Notes On Missing Dependencies in above instructions</summary>
+Dependencies may change every now and then. (i.e. Inscape 1.4 will require tinycss2). To avoid duplication of effort, try using `pip install -r requirements.txt` to update required. This also helps keeping instructions more platform agnostic (except for GTK and matplotlib which are better managed by the distro)
+</details>
```
Comment (t0b3) [inline README.md]: drop this details... as this installation step is already included in above instructions. that was more reasoning for you to adapt requirements.txt instead, which you did :+1:

**Review by mofosyne — COMMENTED** [review]
All minor changes made as requested

**Review by mofosyne — COMMENTED** [review]
Done

File: README.md (hunk 1 above)
Comment (t0b3) [reply README.md:35]: @mofosyne you probably overlooked this

**Comment by t0b3** [comment]
@mofosyne THX for your contributions :rocket: that's what makes opensource so wonderful :smile:

**Comment by mofosyne** [comment]
Not a problem. I would say @miLORD1337 get a lot of credit for getting the ball rolling in the first place.

**Pull request merged**